cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/ProgramRunStatusMonitorServiceTest.java

```diff
@@ -0,0 +1,248 @@
+import io.cdap.cdap.proto.id.ProfileId;
+import io.cdap.cdap.proto.id.ProgramId;
+import io.cdap.cdap.proto.id.ProgramRunId;
+import org.apache.twill.api.RunId;
+import org.apache.twill.common.Cancellable;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.annotation.Nullable;
+
+
+/**
+ * Unit test for {@link ProgramRunStatusMonitorService}
+ */
+public class ProgramRunStatusMonitorServiceTest extends AppFabricTestBase {
+  private static Store store;
+  private static CConfiguration cConf;
+
+  @BeforeClass
+  public static void setup() {
+    store = getInjector().getInstance(DefaultStore.class);
+    cConf = getInjector().getInstance(CConfiguration.class);
+  }
+
+  @After
+  public void clearStore() {
+    store.removeAll(NamespaceId.DEFAULT);
+  }
+
+  @Test
+  public void testStoppingProgramsBeyondTerminateTimeAreKilled() {
+    AtomicInteger sourceId = new AtomicInteger(0);
+    ArtifactId artifactId = NamespaceId.DEFAULT.artifact("testArtifact", "1.0").toApiArtifactId();
+    // set up a workflow for a program in Stopping state
+    Map<String, String> wfSystemArg = ImmutableMap.of(
+      ProgramOptionConstants.CLUSTER_MODE, ClusterMode.ISOLATED.name(),
+      SystemArguments.PROFILE_NAME, ProfileId.NATIVE.getScopedName());
+    ProgramRunId wfId = NamespaceId.DEFAULT.app("test").workflow("testWF").run(randomRunId());
+    store.setProvisioning(wfId, Collections.emptyMap(), wfSystemArg,
+                          Bytes.toBytes(sourceId.getAndIncrement()), artifactId);
+    store.setProvisioned(wfId, 0, Bytes.toBytes(sourceId.getAndIncrement()));
+    store.setStart(wfId, null, Collections.emptyMap(), Bytes.toBytes(sourceId.getAndIncrement()));
+    store.setRunning(wfId, TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()), null,
+                     Bytes.toBytes(sourceId.getAndIncrement()));
+    long currentTimeInSecs = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());
+    store.setStopping(wfId, Bytes.toBytes(sourceId.getAndIncrement()), currentTimeInSecs, currentTimeInSecs);
+    CountDownLatch latch = new CountDownLatch(1);
+    ProgramRuntimeService testService = new AbstractProgramRuntimeService(
+      cConf, null, null, new NoOpProgramStateWriter(), null) {
+      @Nullable
+      public RuntimeInfo lookup(ProgramId programId, RunId runId) {
+        return getRuntimeInfo(programId, latch);
+      }
+    };
+    ProgramRunStatusMonitorService programRunStatusMonitorService
+      = new ProgramRunStatusMonitorService(cConf, store, testService, 5);
+    Assert.assertEquals("Latch count before stop is executed is 1", 1, latch.getCount());
```

Comment:
The message is what's displayed when the assert fails, it should be a message about what's wrong and not a message about what's expected.

In these cases I think it's ok to not give a custom message and let Assert do the default behavior, which will print both the expected and actual values.